apisix/runner/http/request.py

```diff
@@ -0,0 +1,224 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import json
+import apisix.runner.plugin.core as RunnerPlugin
+import apisix.runner.http.method as RunnerMethod
+from apisix.runner.http.protocol import RPC_HTTP_REQ_CALL
+from apisix.runner.http.protocol import RPC_PREPARE_CONF
+from a6pluginproto.HTTPReqCall import Req as A6HTTPReqCallReq
+from a6pluginproto.PrepareConf import Req as A6PrepareConfReq
+
+
+class Request:
+
+    def __init__(self, ty: int = 0, buf: bytes = b''):
+        """
+        Init and parse request
+        :param ty:
+            rpc request protocol type
+        :param buf:
+            rpc request buffer data
+        """
+        self._rpc_type = ty
+        self._rpc_buf = buf
+        self._req_id = 0
+        self._req_conf_token = 0
+        self._req_method = ""
+        self._req_path = ""
+        self._req_headers = {}
+        self._req_configs = {}
+        self._req_args = {}
+        self._req_src_ip = ""
+        self._init()
+
+    @property
+    def rpc_type(self) -> int:
+        return self._rpc_type
+
+    @rpc_type.setter
+    def rpc_type(self, rpc_type: int) -> None:
+        self._rpc_type = rpc_type
+
+    @property
+    def rpc_buf(self) -> bytes:
+        return self._rpc_buf
+
+    @rpc_buf.setter
+    def rpc_buf(self, rpc_buf: bytes) -> None:
+        self._rpc_buf = rpc_buf
+
+    @property
+    def conf_token(self) -> int:
+        return self._req_conf_token
+
+    @conf_token.setter
+    def conf_token(self, req_conf_token: int) -> None:
+        self._req_conf_token = req_conf_token
+
+    @property
+    def id(self) -> int:
+        return self._req_id
+
+    @id.setter
+    def id(self, req_id: int) -> None:
```

Comment:
Do we need to change the id after creating the request?

Reply:
There is no need to modify it at the moment, but I think it is necessary to keep the ID setting API for subsequent expansion